contracts/proposals/ERC1046/TokenMetadata.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.21;
+
+import "../../token/ERC20/ERC20.sol";
+
+
+/**
+ * @title ERC-1047 Token Metadata
+ * @dev See https://eips.ethereum.org/EIPS/eip-1046
+ * @dev tokenURI must respond with a URI that implements https://eips.ethereum.org/EIPS/eip-1047
+ * @dev TODO - update https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC721/ERC721.sol#L17 when 1046 is finalized
+ */
+contract ERC20TokenMetadata is ERC20 {
+  function tokenURI() public view returns (string);
```

Comment:
Note that this interface is `public` here rather than `external` as in the draft.

Are the minor gas savings worth it? https://ethereum.stackexchange.com/questions/19380/external-vs-public-best-practices